ts/parts-3d/Column.ts

```diff
@@ -533,18 +543,42 @@ wrap(H.StackItem.prototype, 'getStackBox', function (
     chart: Highcharts.Chart
 ): void { // #3946
     var stackBox = proceed.apply(this, [].slice.call(arguments, 1));
-
-    // Only do this for 3D chart.
+    // Only do this for 3D column and iherited series.
     if (chart.is3d()) {
-        var pos = ({
-            x: stackBox.x,
-            y: stackBox.y,
-            z: 0
+        var columnSeries: Highcharts.ColumnSeries | undefined;
+        // Check if any series iherits from column 3D
+        chart.series.forEach(function (
+            series: Highcharts.Series
+        ): void {
+            if (series instanceof seriesTypes.column && series.data.length) {
+                columnSeries = series;
+            }
         });
+        // If any series is a column series, use its barW, z and depth
+        // parameters for correct stackLabels position calculation
+        if (columnSeries) {
+            var height: number = arguments[6],
```

Comment:
`var` in here doesn't look good - consider using `let` / `const` or placing all `var` at the top of the scope